pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -939,37 +708,55 @@ def inputs_for_chunk(self, chunk_key: ChunkKey) -> Sequence[InputKey]:
         ninputs = self.file_pattern.dims[self.file_pattern.concat_dims[0]]
         return inputs_for_chunk(chunk_key, self.inputs_per_chunk, ninputs)
 
+
+@dataclass
+class XarrayZarrRecipe(BaseRecipe):
+
+    config: XarrayZarrRecipeConfig
```

Comment:
If you didn't want to break the previous API: The `BaseRecipe` could take a constructor that builds the config -- that way, the use of configs would be more of an internal implementation detail.

Reply:
Some thoughts on the interface for this: 
Since python doesn't support multiple constructors, you could structure the API like 
```
def __init__(self, paramA=None, paramB=None, ..., *, config: t.Optional[XarrayZarrRecipeConfig] = None):
```
This way, the config would have to be passed in as a key-word arg.